nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/bigquery/PutBigQueryBatch.java

```diff
@@ -0,0 +1,269 @@
+        @WritesAttribute(attribute = BigQueryAttributes.CREATE_DISPOSITION_ATTR, description = BigQueryAttributes.CREATE_DISPOSITION_DESC),
+        @WritesAttribute(attribute = BigQueryAttributes.WRITE_DISPOSITION_ATTR, description = BigQueryAttributes.WRITE_DISPOSITION_DESC),
+        @WritesAttribute(attribute = BigQueryAttributes.MAX_BADRECORDS_ATTR, description = BigQueryAttributes.MAX_BADRECORDS_DESC),
+        @WritesAttribute(attribute = BigQueryAttributes.JOB_CREATE_TIME_ATTR, description = BigQueryAttributes.JOB_CREATE_TIME_DESC),
+        @WritesAttribute(attribute = BigQueryAttributes.JOB_END_TIME_ATTR, description = BigQueryAttributes.JOB_END_TIME_DESC),
+        @WritesAttribute(attribute = BigQueryAttributes.JOB_START_TIME_ATTR, description = BigQueryAttributes.JOB_START_TIME_DESC),
+        @WritesAttribute(attribute = BigQueryAttributes.JOB_LINK_ATTR, description = BigQueryAttributes.JOB_LINK_DESC),
+        @WritesAttribute(attribute = BigQueryAttributes.JOB_ERROR_MSG_ATTR, description = BigQueryAttributes.JOB_ERROR_MSG_DESC),
+        @WritesAttribute(attribute = BigQueryAttributes.JOB_ERROR_REASON_ATTR, description = BigQueryAttributes.JOB_ERROR_REASON_DESC),
+        @WritesAttribute(attribute = BigQueryAttributes.JOB_ERROR_LOCATION_ATTR, description = BigQueryAttributes.JOB_ERROR_LOCATION_DESC)
+})
+
+public class PutBigQueryBatch extends AbstractBigQueryProcessor {
+
+    public static final PropertyDescriptor SOURCE_TYPE = new PropertyDescriptor
+            .Builder().name(BigQueryAttributes.SOURCE_TYPE_ATTR)
+            .displayName("Load file type")
+            .description(BigQueryAttributes.SOURCE_TYPE_DESC)
+            .required(true)
+            .allowableValues(FormatOptions.json().getType(), FormatOptions.avro().getType(), FormatOptions.csv().getType())
+            .defaultValue(FormatOptions.avro().getType())
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor IGNORE_UNKNOWN = new PropertyDescriptor.Builder()
+            .name(BigQueryAttributes.IGNORE_UNKNOWN_ATTR)
+            .displayName("Ignore Unknown Values")
+            .description(BigQueryAttributes.IGNORE_UNKNOWN_DESC)
+            .required(true)
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .build();
+
+    public static final PropertyDescriptor CREATE_DISPOSITION = new PropertyDescriptor.Builder()
+            .name(BigQueryAttributes.CREATE_DISPOSITION_ATTR)
+            .displayName("Create Disposition")
+            .description(BigQueryAttributes.CREATE_DISPOSITION_DESC)
+            .required(true)
+            .allowableValues(JobInfo.CreateDisposition.CREATE_IF_NEEDED.name(), JobInfo.CreateDisposition.CREATE_NEVER.name())
```

Comment:
Can we use the AllowableValue object to add a description for each possible value? That would be super useful.

Something like:

````java
    static final AllowableValue CREATE_IF_NEEDED = new AllowableValue("CREATE_IF_NEEDED", JobInfo.CreateDisposition.CREATE_IF_NEEDED.name(), "Configures the job to create the table if it does not exist.");
````

I'm retrieving the description from the source code.